### PR TITLE
ISPN-15200 RENAME RENAMENX 

### PR DESCRIPTION
**Pull request by rigazilla**
RENAME RENAMENX commands

https://issues.redhat.com/browse/ISPN-15200

~TODO: missing tests on different data types~ added, comment below

### CODE REVIEW

**Comment by rigazilla** [comment]
I added a fix to some general commands to handle non-string type

File: documentation/src/main/asciidoc/topics/ref_redis_commands.adoc [hunk 1]
```diff
@@ -255,6 +255,12 @@ provides a read-uncommitted isolation.
 | https://redis.io/commands/readwrite[READWRITE]
 |
 
+| https://redis.io/commands/readwrite[RENAME]
+|
+
+| https://redis.io/commands/readwrite[RENAMENX]
```
Comment (karesti) [inline documentation/src/main/asciidoc/topics/ref_redis_commands.adoc:261]: docs link

File: documentation/src/main/asciidoc/topics/ref_redis_commands.adoc [hunk 2]
```diff
@@ -255,6 +255,12 @@ provides a read-uncommitted isolation.
 | https://redis.io/commands/readwrite[READWRITE]
 |
 
+| https://redis.io/commands/readwrite[RENAME]
```
Comment (karesti) [inline documentation/src/main/asciidoc/topics/ref_redis_commands.adoc:258]: docs link

File: server/resp/src/main/java/org/infinispan/server/resp/commands/generic/RENAME.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+   public static CompletionStage<RespRequestHandler> rename(Resp3Handler handler, byte[] srcKey, byte[] dstKey,
+         ChannelHandlerContext ctx, BiConsumer<?, ByteBufPool> consumer) {
+      MediaType vmt = handler.cache().getValueDataConversion().getStorageMediaType();
+      final AdvancedCache<byte[], Object> acm = handler.cache().<byte[],Object>withMediaType(MediaType.APPLICATION_OCTET_STREAM, vmt);
+      return acm.removeAsyncEntry(srcKey).thenCompose(e -> {
+         if (e == null) {
+            RespErrorUtil.noSuchKey(handler.allocator());
+            return handler.myStage();
+         } else {
```
Comment (karesti) [inline server/resp/src/main/java/org/infinispan/server/resp/commands/generic/RENAME.java:49]: no need else here

File: server/resp/src/main/java/org/infinispan/server/resp/commands/generic/RENAME.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+            var timeService = handler.respServer().getTimeService();
+            // Immortal entry, copy metadata
+            if (e.getLifespan()<=0) {
+               var newMeta = e.getMetadata().builder();
+               return handler.stageToReturn(acm.putAsyncEntry(dstKey, e.getValue(), newMeta.build()).thenApply(ignore -> 1L), ctx,
```
Comment (karesti) [inline server/resp/src/main/java/org/infinispan/server/resp/commands/generic/RENAME.java:54]: @rigazilla if the put fails after the remove we might lose the data, so I guess this is something we need to consider since we are unable to rename keys in a single op

File: server/resp/src/main/java/org/infinispan/server/resp/commands/generic/RENAME.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+                     (BiConsumer<Object, ByteBufPool>)consumer);
+            }
+            // Otherwise copy metadata and preserve lifespan or do nothing if expired
+            long newLifespan = e.getLifespan() + e.getCreated() - timeService.wallClockTime();
+            if (newLifespan > 0) {
+               var newMeta = e.getMetadata().builder().lifespan(newLifespan);
+               return handler.stageToReturn(acm.putAsyncEntry(dstKey, e.getValue(), newMeta.build()).thenApply(ignore -> 1L), ctx,
```
Comment (karesti) [inline server/resp/src/main/java/org/infinispan/server/resp/commands/generic/RENAME.java:61]: you can to a single return of stageToReturn instead of the 3

File: server/resp/src/test/java/org/infinispan/server/resp/RespSingleNodeTest.java [hunk 6]
```diff
@@ -735,6 +736,17 @@ public void testPTTL() {
       assertThat(redis.pttl(k(2))).isEqualTo(10_000_000L);
    }
 
+   @Test
+   public void testPTTLTypes() {
+      RedisCommands<String, String> redis = redisConnection.sync();
```
Comment (karesti) [inline server/resp/src/test/java/org/infinispan/server/resp/RespSingleNodeTest.java:741]: is this a missing test or something that does not belong in this PR

File: server/resp/src/test/java/org/infinispan/server/resp/RespSingleNodeTest.java [hunk 7]
```diff
@@ -835,4 +847,73 @@ public void testExpireAt() {
...
+
+   @Test
+   public void testRename() {
+      RedisCommands<String, String> redis = redisConnection.sync();
+      var srcKey = k(0);
+      var dstKey = k(1);
+      var val = v();
+      redis.set(srcKey, val);
+      redis.rename(srcKey, dstKey);
+      assertThat(redis.get(dstKey)).isEqualTo(val);
```
Comment (karesti) [inline server/resp/src/test/java/org/infinispan/server/resp/RespSingleNodeTest.java:859]: test unexisting key missing

**Review by karesti — CHANGES_REQUESTED** [review]

File: server/resp/src/test/java/org/infinispan/server/resp/RespSingleNodeTest.java (hunk 7 above)
Comment (rigazilla) [reply server/resp/src/test/java/org/infinispan/server/resp/RespSingleNodeTest.java:859]: ok added

File: server/resp/src/main/java/org/infinispan/server/resp/commands/generic/RENAME.java (hunk 4 above)
Comment (rigazilla) [reply server/resp/src/main/java/org/infinispan/server/resp/commands/generic/RENAME.java]: I added a warning in the doc comments

**Comment by karesti** [comment]
@rigazilla for me this is fine, squash in a single commit 🙏🏻

**Comment by rigazilla** [comment]
> @rigazilla for me this is fine, squash in a single commit 🙏🏻

ok done
